homeassistant/components/crownstone/__init__.py

```diff
@@ -0,0 +1,34 @@
+"""Integration for Crownstone."""
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .hub import CrownstoneHub
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Initiate the Crownstone component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Initiate the Crownstone API login from a config entry."""
+    crownstone_hub = CrownstoneHub(hass, entry)
+
+    if not await crownstone_hub.async_setup():
+        return False
```

Comment:
Should this abort or try again later?
In case it should try again later, ConfigEntryNotReady should be raised.

Reply:
Within the `async_setup` method, i already raise ConfigEntryNotReady upon an unknown error, however i don't know if this is the correct usage
```python
except CrownstoneUnknownError:
        _LOGGER.error("Unknown error during login")
        raise ConfigEntryNotReady
```
My idea was, if it's an unknown error it could be caused by the server and should be retried later. If `async_setup` returns `False` something must have gone wrong during authentication:
```python
except CrownstoneAuthenticationError as auth_err:
        _LOGGER.error(
           "Auth error during login with type: %s and message: %s",
           auth_err.type,
           auth_err.message,
        )
        return False
```
Which is quite unlikely, but in that case it should abort and the user should re-authenticate.